### PR TITLE
[BugFix]: Fix `sglang.bench_one_batch`

### PR DESCRIPTION
**Pull request by ljw-mc**
## Motivation

On the main branch, model final logits from the following commands do not match.

- `python3 scripts/playground/reference_hf.py --model-path meta-llama/Llama-3.1-8B --dtype bfloat16`

- `python3 -m sglang.bench_one_batch --correctness-test --model meta-llama/Llama-3.1-8B` 

The reason is one line of code casts `req.prefix_indices` from `torch.int64` to `torch.int32`, which is not what the `write_req_to_token_pool_triton` kernel expects.

## Before change:
```python
➜  sglang git:(main) python3 -m sglang.bench_one_batch --correctness-test --model meta-llama/Llama-3.1-8B
[2026-01-12 07:10:52 TP0] Init torch distributed begin.
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[2026-01-12 07:10:52 TP0] Init torch distributed ends. mem usage=0.00 GB
[2026-01-12 07:10:52 TP0] MOE_RUNNER_BACKEND is not initialized, the backend will be automatically selected
[2026-01-12 07:10:53 TP0] Ignore import error when loading sglang.srt.models.glmasr: cannot import name 'GlmAsrConfig' from 'transformers' (/usr/local/lib/python3.12/dist-packages/transformers/__init__.py)
[2026-01-12 07:10:54 TP0] Load weight begin. avail mem=21.84 GB
[2026-01-12 07:10:54 TP0] Found local HF snapshot for meta-llama/Llama-3.1-8B at /home/devuser/.cache/huggingface/hub/models--meta-llama--Llama-3.1-8B/snapshots/d04e592bb4f6aa9cfee91e2e20afa771667e1d4b; skipping download.
Loading safetensors checkpoint shards:   0% Completed | 0/4 [00:00<?, ?it/s]
Loading safetensors checkpoint shards:  25% Completed | 1/4 [00:02<00:08,  2.92s/it]
Loading safetensors checkpoint shards:  50% Completed | 2/4 [00:03<00:02,  1.39s/it]
Loading safetensors checkpoint shards:  75% Completed | 3/4 [00:04<00:01,  1.11s/it]
Loading safetensors checkpoint shards: 100% Completed | 4/4 [00:40<00:00, 14.93s/it]
Loading safetensors checkpoint shards: 100% Completed | 4/4 [00:40<00:00, 10.03s/it]

[2026-01-12 07:11:34 TP0] Load weight end. type=LlamaForCausalLM, dtype=torch.bfloat16, avail mem=6.79 GB, mem usage=15.05 GB.
[2026-01-12 07:11:34 TP0] Using KV cache dtype: torch.bfloat16
[2026-01-12 07:11:34 TP0] KV Cache is allocated. #tokens: 26471, K size: 1.62 GB, V size: 1.62 GB
[2026-01-12 07:11:34 TP0] Memory pool end. avail mem=2.52 GB
[2026-01-12 07:11:35 TP0] Capture cuda graph begin. This can take up to several minutes. avail mem=2.01 GB
[2026-01-12 07:11:35 TP0] Capture cuda graph bs [1, 2, 4, 8, 12, 16, 24]
Capturing batches (bs=1 avail_mem=1.87 GB): 100%|████████████████████████████████| 7/7 [00:03<00:00,  1.96it/s]
[2026-01-12 07:11:39 TP0] Capture cuda graph end. Time elapsed: 4.12 s. mem usage=0.15 GB. avail mem=1.86 GB.
max_total_num_tokens=26471

input_ids=[[128000, 791, 6864, 315, 9822, 374], [128000, 791, 6864, 315, 279, 3723, 17262, 316, 374], [128000, 15724, 374, 264, 40798, 1938, 323, 358, 1093]]

prefill logits (first half): tensor([[ 1.5938,  1.7812, -0.8398,  ..., -5.5625, -5.5625, -5.5625],
        [ 1.5938,  1.7812, -0.8398,  ..., -5.5625, -5.5625, -5.5625],
        [ 2.0938,  2.6094,  2.0156,  ..., -8.1875, -8.1875, -8.1875]],
       device='cuda:0') 

prefill logits (final): tensor([[ 5.9062,  2.8438,  0.8633,  ..., -7.1562, -7.1562, -7.1562],
        [ 4.5625,  3.7969,  0.7734,  ..., -7.5000, -7.5000, -7.5000],
        [ 8.1875,  2.0000,  0.6680,  ..., -8.8750, -8.8750, -8.8750]],
       device='cuda:0') 

========== Prompt 0 ==========
<|begin_of_text|>The capital of France is a city of art and culture, and it is also a city of fashion. 

========== Prompt 1 ==========
<|begin_of_text|>The capital of the United Kindom is a city of contrasts. It is a city of history and tradition, but also 

========== Prompt 2 ==========
<|begin_of_text|>Today is a sunny day and I like to go out for a walk. I am going to the park. I am 

[rank0]:[W112 07:11:48.384260424 ProcessGroupNCCL.cpp:1524] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
```

## After change:
```python
➜  sglang git:(ljw-mc/fix_bench_one_batch) python3 -m sglang.bench_one_batch --correctness-test --model meta-llama/Llama-3.1-8B            
[2026-01-12 07:03:04 TP0] Init torch distributed begin.
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[2026-01-12 07:03:04 TP0] Init torch distributed ends. mem usage=0.00 GB
[2026-01-12 07:03:04 TP0] MOE_RUNNER_BACKEND is not initialized, the backend will be automatically selected
[2026-01-12 07:03:05 TP0] Ignore import error when loading sglang.srt.models.glmasr: cannot import name 'GlmAsrConfig' from 'transformers' (/usr/local/lib/python3.12/dist-packages/transformers/__init__.py)
[2026-01-12 07:03:06 TP0] Load weight begin. avail mem=21.84 GB
[2026-01-12 07:03:06 TP0] Found local HF snapshot for meta-llama/Llama-3.1-8B at /home/devuser/.cache/huggingface/hub/models--meta-llama--Llama-3.1-8B/snapshots/d04e592bb4f6aa9cfee91e2e20afa771667e1d4b; skipping download.
Loading safetensors checkpoint shards:   0% Completed | 0/4 [00:00<?, ?it/s]
Loading safetensors checkpoint shards:  25% Completed | 1/4 [00:32<01:37, 32.59s/it]
Loading safetensors checkpoint shards:  50% Completed | 2/4 [00:41<00:37, 18.59s/it]
Loading safetensors checkpoint shards:  75% Completed | 3/4 [01:18<00:27, 27.21s/it]
Loading safetensors checkpoint shards: 100% Completed | 4/4 [01:56<00:00, 31.15s/it]
Loading safetensors checkpoint shards: 100% Completed | 4/4 [01:56<00:00, 29.01s/it]

[2026-01-12 07:05:02 TP0] Load weight end. type=LlamaForCausalLM, dtype=torch.bfloat16, avail mem=6.79 GB, mem usage=15.05 GB.
[2026-01-12 07:05:02 TP0] Using KV cache dtype: torch.bfloat16
[2026-01-12 07:05:02 TP0] KV Cache is allocated. #tokens: 26471, K size: 1.62 GB, V size: 1.62 GB
[2026-01-12 07:05:02 TP0] Memory pool end. avail mem=2.52 GB
[2026-01-12 07:05:03 TP0] Capture cuda graph begin. This can take up to several minutes. avail mem=2.01 GB
[2026-01-12 07:05:03 TP0] Capture cuda graph bs [1, 2, 4, 8, 12, 16, 24]
Capturing batches (bs=1 avail_mem=1.87 GB): 100%|████████████████████████████████| 7/7 [00:03<00:00,  1.84it/s]
[2026-01-12 07:05:07 TP0] Capture cuda graph end. Time elapsed: 4.40 s. mem usage=0.15 GB. avail mem=1.86 GB.
max_total_num_tokens=26471

input_ids=[[128000, 791, 6864, 315, 9822, 374], [128000, 791, 6864, 315, 279, 3723, 17262, 316, 374], [128000, 15724, 374, 264, 40798, 1938, 323, 358, 1093]]

prefill logits (first half): tensor([[ 1.5938,  1.7812, -0.8398,  ..., -5.5625, -5.5625, -5.5625],
        [ 1.5938,  1.7812, -0.8398,  ..., -5.5625, -5.5625, -5.5625],
        [ 2.0938,  2.6094,  2.0156,  ..., -8.1875, -8.1875, -8.1875]],
       device='cuda:0') 

prefill logits (final): tensor([[ 4.8125,  3.1719,  1.1016,  ..., -6.7188, -6.7188, -6.7188],
        [ 4.6562,  3.5156,  1.4922,  ..., -8.3125, -8.3125, -8.3125],
        [ 9.3750,  2.7500,  1.2656,  ..., -7.3750, -7.3750, -7.3750]],
       device='cuda:0') 

========== Prompt 0 ==========
<|begin_of_text|>The capital of France is a city of many faces. It is a city of history, a city of 

========== Prompt 1 ==========
<|begin_of_text|>The capital of the United Kindom is London. It is the largest city in the UK and the largest city in the 

========== Prompt 2 ==========
<|begin_of_text|>Today is a sunny day and I like to go out for a walk. I am going to the park. I am 

[rank0]:[W112 07:05:18.129321178 ProcessGroupNCCL.cpp:1524] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
```

## Hugging Face Reference
```python
➜  sglang git:(main) python3 scripts/playground/reference_hf.py --model-path meta-llama/Llama-3.1-8B --dtype bfloat16
`torch_dtype` is deprecated! Use `dtype` instead!
Loading checkpoint shards: 100%|█████████████████████████████████████████████████| 4/4 [01:19<00:00, 19.94s/it]
The following generation flags are not valid and may be ignored: ['temperature', 'top_p']. Set `TRANSFORMERS_VERBOSITY=info` for more details.
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:128001 for open-end generation.
The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.

========== Prompt 0 ==========
prefill logits (final) tensor([ 4.8125,  3.1562,  1.0938,  ..., -6.7188, -6.7188, -6.7188],
       device='cuda:0', dtype=torch.bfloat16)
<|begin_of_text|>The capital of France is a city of many faces. It is a city of history, a city of
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:128001 for open-end generation.

========== Prompt 1 ==========
prefill logits (final) tensor([ 4.6562,  3.5000,  1.4688,  ..., -8.3125, -8.3125, -8.3125],
       device='cuda:0', dtype=torch.bfloat16)
<|begin_of_text|>The capital of the United Kindom is London. It is the largest city in the UK and the largest city in the
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:128001 for open-end generation.

========== Prompt 2 ==========
prefill logits (final) tensor([ 9.5000,  2.8125,  1.3203,  ..., -7.2812, -7.2812, -7.2812],
       device='cuda:0', dtype=torch.bfloat16)
<|begin_of_text|>Today is a sunny day and I like to go out for a walk. I am going to the park. I am
```

## Checklist

- [X] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [X] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Review by JustinTong0323 — APPROVED** [review]

**Comment by yuan-luo** [comment]
Please also consider ViT CUDA Graph feature.

**Comment by ljw-mc** [comment]
> Please also consider ViT CUDA Graph feature.

This PR fixes a bug inside `sglang.bench_one_batch`, which tests text-only LLMs.  It shouldn't affect ViTs/VLMs.

@yuan-luo Can you elaborate further on this? Maybe we should scope this for another issue or PR?
